msal/application.py

```diff
@@ -131,6 +131,14 @@ def __init__(
             It will be passed to the
             `timeout parameter in the underlying requests library
             <http://docs.python-requests.org/en/v2.9.1/user/advanced/#timeouts>`_
+        :param app_name: (optional)
+            You can provide your application name for telemetry purposes which will be
+            passed as an additional header value "x-app-name"
+            By default, the additional header value will not be passed
+        :param app_version: (optional)
```

Comment:
1. Given that this feature requirement is an SDK consistency work item, shall we do a little research on a couple other MSAL's interface on this? We can paste their API in this PR's description as a comparison.

2. In particular, given that the underlying wire parameter happens to be named `x-app-ver`, is this parameter named `app_version` or `app_ver` in other MSALs? Want to double check before our merge and release, because we won't be able to easily change a parameter's name without a breaking change.

3. To a lesser extent, how is this feature documented in other MSALs' inline API doc? These 2 parameters need app developer's opt-in, and we currently advertise them as "for telemetry purposes", but app developers might not immediately realize that they don't actually have access to those telemetry data. Perhaps we can somehow reword this to say something like:

            You can provide your application version for Microsoft telemetry purposes.
            Default value is None, means it will not be passed to Microsoft.

Reply:
Thanks @rayluo , All these are really good points to note.

1. I have added the MSAL Java Interface in the description. 
2. The Java implementation uses applicationName and applicationVersion. So I think its just more informative to use it as `version` instead of just `ver`.
3. The exact documentation for Java is this:
```
/**
         * Sets application version for telemetry purposes
         * 
         * @param val application version
         * @return instance of the Builder on which method was called
         */
```
But I think your point is valid and we can mention it explicitly that its for Microsoft telemetry purposes. But then if I understand it correct, they can actually use it in their own telemetry usecases too if they want to and are logging these requests on their side. Let me know your thoughts.